packages/shared/src/variants/baduk.ts

```diff
@@ -206,3 +230,16 @@ export function groupHasLiberties(group: CoordinateLike[], board: Grid<Color>) {
 function count_color<T>(value: T) {
   return (total: number, color: T) => total + (color === value ? 1 : 0);
 }
+
+export type GridBadukConfig = {
+  width: number;
+  height: number;
+  komi: number;
+  board?: GridBoardConfig;
+};
```

Comment:
Can we use something like
```
export type GridBadukConfig = {
  width: number;
  height: number;
  komi: number;
} | {
  board: GridBoardConfig;
  komi: number;
}
```

otherwise there's room for trouble:
```
{
  komi: 6.5;
  width: 9;
  height: 9;
  board: {
    type: "grid";
    width: 19;
    height: 19;
  }
}
```

Reply:
done